### ISSUE TITLE
cache clean command to remove temporary files

### ISSUE THREAD

**Issue by alexchandel** [chocolatey/choco#1124]
chocolately creates lots of temporary/cached files when installing packages, and these aren't removed when the installation is complete. Even older version aren't removed. Chocolately needs a `cleanup` command that tells it to delete all its temporary and cached files.

**Editor's note:** The original issue was filed at https://github.com/chocolatey/chocolatey/issues/253

**Comment by ferventcoder** [comment]
Duplicate

**Comment by ferventcoder** [comment]
So I thought this was a dupe - we've discussed this and have been planning for awhile. I'm surprised there is no actual issue filed if there is not already.

**Comment by ferventcoder** [comment]
> Older versions are not removed

Please elaborate

**Comment by ferventcoder** [comment]
The lot of temp files are cleaned up - change your cacheLocation config value. Then temp is more deterministic and things are cleaned up better

**Comment by alexchandel** [comment]
> Please elaborate

I just did `choco upgrade all`, and the cached installers for the previous versions of the upgraded packages weren't deleted.

**Comment by ferventcoder** [comment]
>cached installers for the previous versions 

Ah, cached installers. Not package versions left behind.

**Comment by alexchandel** [comment]
Right, they can be hundreds of megabytes, and scattered across dozens of users' temp folders.

Also why not use a global location for cached installers if chocolatey is installed globally? Or better yet, once downloaded, move the cached installers to a cache folder in chocolatey's folder?

Either way a `choco cleanup` command would be nice so normal chocolatey users don't have to route around in their temp folders.

**Comment by ferventcoder** [comment]
Re: Global location - did you miss my response on setting a global temp location by setting the cacheLocation?

**Comment by ferventcoder** [comment]
Found it - https://github.com/chocolatey/chocolatey/issues/253

**Comment by 07416** [comment]
I have 2 GB of cache files. Is it safe to delete everything manually? 
If I need to reinstall a program I can download it again.

**Comment by ferventcoder** [comment]
@07416 yes, apologies if that is not clear. As long as you are not saying you have 2GB of files including nupkg files in what you consider cache - the "%ChocolateyInstall%\lib" location is not what we would consider cache in this case.

**Comment by 07416** [comment]
@ferventcoder The location is `~\AppData\Local\Temp\chocolatey`.

**Comment by ferventcoder** [comment]
Yep, that's fine.

**Comment by mitchmenghi** [comment]
Any update on this im a new user and also noticed that all temp files are there, if they are not needed after an install success there should be a switch to delete or better still some sort of a clean command... :)

**Comment by ferventcoder** [comment]
@mitchmenghi This was implemented in Package Reducer, which also comes with `choco optimize` - https://chocolatey.org/docs/features-package-reducer

This is still being considered.

**Comment by ferventcoder** [comment]
This is on the backlog apologies.

**Comment by bcurran3** [comment]
You can also look at https://chocolatey.org/packages/choco-cleaner/0.0.1
(Not intentionally trying to undermine you Rob!)

**Comment by aaccioly** [comment]
I'm glad that I've come across this issue. I'm testing Chocolatey and my `~\AppData\Local\Temp\chocolatey` had several GBs of cached installers after one day of use. Quite a few of those files were duplicated in `lib`. @ferventcoder, I understand that automatic package reduction is a killer feature of the licensed version, however would you please consider releasing at least the cache cleaning command on the community edition?

I'll be glad to buy the pro version or donate to help you keep the good work in Chocolatey. However - and please take this as constructive feedback - all major package managers that I can think of have some kind of `clean` command (apt, yum, zypper, homebrew, etc). This is basic functionality, I don't think it should be a paid feature.

**Comment by N3X15** [comment]
Really?  This is a premium feature?

**Comment by ghost** [comment]
> 
> 
> You can also look at https://chocolatey.org/packages/choco-cleaner/0.0.1

Why does that package show as rejected?

**Comment by bcurran3** [comment]
@07416 

I'm glad you brought that up!
v0.0.1 will always show as rejected.
v0.0.2+ is live and well, current version is 0.0.5.1
That link points ONLY to v0.0.1

Anyone in the future reading this thread, please click the following link to jump directly to the current release:
https://chocolatey.org/packages/choco-cleaner

**Comment by black-snow** [comment]
Looking forward to having this feature in the free version.

**Comment by TheNavyBear** [comment]
I agree with others here. The paid version should have features the free version doesn't, otherwise no one would used the paid version and funding would dry up. However I also feel that cache cleanup is a basic part of any package manager. Maybe a good middle ground would be to have basic cache cleanup (clearing the temp folder) in the free version, and to have all the other reducer/optimizer stuff (reducing NuPKG size, etc) continue being a paid feature. Basic cache cleanup is good for users and it can go a long way in making Chocolatey more favorable in the eyes of users. That's just my two cents.

I thank you all for all the work you do on Chocolatey, and I appreciate having Chocolatey to use. I will always continue to support y'all anytime I can afford to, and I hope y'all keep up the great work!
